inference-engine/src/vpu/common/src/ngraph/operations/dynamic_shape_resolver.cpp

```diff
@@ -30,18 +34,159 @@ void DynamicShapeResolver::validate_and_infer_types() {
+    if (m_mode == DynamicShapeResolverMode::INFER_UPPER_BOUND_SHAPE) {
+        NODE_VALIDATION_CHECK(this, get_input_partial_shape(0).is_static(), "(", get_friendly_name(), ") does not support dynamic shape for data tensor");
+
+        const auto& dataShape = get_input_shape(0);
+        NODE_VALIDATION_CHECK(this, dimsShape.size() == 1 && dimsShape.front() == dataShape.size(), "(", get_friendly_name(), ") inputs shapes mismatch: first "
+            "input shape = ", dataShape, " second input shape = ", dimsShape, " but ", dataShape, " and ", Shape{dataShape.size()}, " are expected");
+
+        set_output_type(0, dataElementType, dataShape);
+    } else {
+        NODE_VALIDATION_CHECK(this, get_input_partial_shape(0).rank() == dimsShape.front(),
+                "(", get_friendly_name(), ") data and shape ranks must be equal, proided: ",
+                get_input_partial_shape(0).rank(), " vs ", dimsShape.front());
+
+        set_output_type(0, dataElementType,
+                        ngraph::PartialShape::dynamic(get_input_partial_shape(0).rank()));
+    }
 }
 
 bool DynamicShapeResolver::visit_attributes(ngraph::AttributeVisitor&) {
     return true;
 }
 
+namespace {
+
+template<element::Type_t ET>
+bool getShapeFromHostTensorData(const HostTensorPtr& data, Shape& result) {
+    using T = typename element_type_traits<ET>::value_type;
+    T *dataPtr = data->get_data_ptr<ET>();
+    if (!dataPtr) {
+        return false;
+    }
+    size_t outputRank = data->get_shape()[0];
+
+    for (int i = 0; i < outputRank; i++) {
+        result.push_back(dataPtr[i]);
+    }
+
+    return true;
+}
+
+bool getShapeFromHostTensorData(const HostTensorPtr& data, Shape& shape) {
+    switch (data->get_element_type()) {
+        case element::Type_t::i8:
+            return getShapeFromHostTensorData<element::Type_t::i8>(data, shape);
+        case element::Type_t::i16:
+            return getShapeFromHostTensorData<element::Type_t::i16>(data, shape);
+        case element::Type_t::i32:
+            return getShapeFromHostTensorData<element::Type_t::i32>(data, shape);
+        case element::Type_t::i64:
+            return getShapeFromHostTensorData<element::Type_t::i64>(data, shape);
+        case element::Type_t::u8:
+            return getShapeFromHostTensorData<element::Type_t::u8>(data, shape);
+        case element::Type_t::u16:
+            return getShapeFromHostTensorData<element::Type_t::u16>(data, shape);
+        case element::Type_t::u32:
+            return getShapeFromHostTensorData<element::Type_t::u32>(data, shape);
+        case element::Type_t::u64:
+            return getShapeFromHostTensorData<element::Type_t::u64>(data, shape);
+        default:
+            return false;
+    }
+    return true;
+}
+
+template<element::Type_t InType>
+bool evaluate(const HostTensorPtr& inputTensor,
+              const HostTensorPtr& outShapeTensor,
+              const HostTensorPtr& outputTensor) {
```

Comment:
Nitpick: looks like both 2 last arguments are outputs, while it's not true

Reply:
Fixed